Makefile.uk

```diff
@@ -0,0 +1,190 @@
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/PcdDecode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/PcxDecode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/PcxEncode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/Point.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/Quant.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/QuantHash.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/QuantHeap.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/QuantOctree.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/QuantPngQuant.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/RankFilter.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/RawDecode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/RawEncode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/Reduce.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/Resample.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/SgiRleDecode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/Storage.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/SunRleDecode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/TgaRleDecode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/TgaRleEncode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/TiffDecode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/Unpack.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/UnpackYCC.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/UnsharpMask.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/XbmDecode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/XbmEncode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/ZipDecode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/ZipEncode.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/libImaging/codec_fd.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/map.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/outline.c
+LIBPYTHON_PILLOW_SRCS-y += $(LIBPYTHON_PILLOW_SRC)/src/path.c
+
+
+################################################################################
+# Root filesystem
+##############################################################################
+
+ifeq ($(CONFIG_LIBPYTHON_PILLOW),y)
+# Install into main python rootfs & cleanup non-python files
+$(PYTHON_ROOTFS)/.pillow_done: $(PYTHON_ROOTFS)/.keep
+	. $(PYTHON_ROOTFS)/bin/activate && cd $(LIBPYTHON_PILLOW_SRC) && pip install . --prefix=$(PYTHON_ROOTFS) --no-compile
+	_dir=`find "$(PYTHON_ROOTFS)" -maxdepth 4 -type d -name PIL`; \
```

Comment:
That depth was specifically 4 because of where PIL gets installed to by default, and my not willing to make this so generic it might interfere with other files in the rootfs (as you can probably see it's not the most robust of `find` invocations and would pick up any dir named `PIL`).
That said, how did you end up installing the pylibs under `rootfs/local` anyway? (and more to the point, _why_ `rootfs/local` instead of the more standard `rootfs/usr/local`)

> I get "No module named PIL__imaging" on testing. Again, might be just something I'm doing wrong on the fs setup.

The actual bindings to the python interpreter are a separate PR for the python lib: https://github.com/unikraft/lib-python3/pull/22. That one provides the native bindings in `modules_config.c` so the interpreter can pick them up.

Reply:
> That said, how did you end up installing the pylibs under rootfs/local anyway? (and more to the point, why rootfs/local instead of the arguably more widespread rootfs/usr/local). If this is something we can likely expect to happen again, there's probably some rework of the rootfs build recipe in order.

Just ran the `make path=rootfs python-rootfs` command, nothing more.

> The actual bindings to the python interpreter are a separate PR for the python lib: https://github.com/unikraft/lib-python3/pull/22. That one provides the native bindings in modules_config.c so the interpreter can pick them up.

Ah yes, I missed that.

Reply:
Mkay, I tried my hand at reproducing this and no dice :/
@StefanJum could you please post your system config (as well as which python/venv you're using to build the rootfs) along with the console output from the rootfs build? I want to figure out in what situations do the pylibs get installed under `/local/` since that's somewhere that the interpreter will not find them under the current config (and therefore will be a nasty surprise to anyone building python).

Reply:
@andreittr I found the problem, the `python3` command was pointing to a different python version (3.8), while the `python` command was pointing to 3.10. Maybe we should use smth like `$(PYTHON)` and `$(PIP)`, that can be set by the user, instead of the literal commands (that is in all python-related repositories). Dealing with multiple versions on your system gets really frustrating.

Especially in the `lib-python3` `Makefile.uk`, that uses [both `python` and `python3`](https://github.com/unikraft/lib-python3/blob/staging/Makefile.uk#L498-L500) commands :grimacing: 

If this sounds out of scope for this PR I'll approve it, everything else works great.

Reply:
@StefanJum I believe virtualenvs are the nearly universal way to select custom interpreters and/or stdlibs in the Python ecosystem and IMO it would be wisest to stick to their conventions instead of introducing our own with `$(PYTHON)` and/or `$(PIP)`. AFAIK virtualenvs provide both `python` and `python3` as interchangeable symllinks to eachother.

That said, even if we did use `$(PYTHON)`/`$(PIP)` for our initial invocation, we'd still have no control over the name by which any subprocess interpreters would be called (since Python/pip doesn't know or care abour `$(PYTHON)`), so ensuring both names point to the same binary is the most robust way of handling this.

Using both `python3` and `python` is an unintended bug in `lib-python3`; it should use just use `python` everywhere. I can push a hotfix commit to https://github.com/unikraft/lib-python3/pull/22 so we have everything consistent for the release.